### PR TITLE
Improvements on test CI, introduction of prettier functional tests and minor bugfixes

### PR DESCRIPTION
**Pull request by kedhammar**

### CODE REVIEW

File: taca/analysis/analysis_element.py [hunk 1]
```diff
@@ -39,7 +39,7 @@ def _process(run):
         sequencing_done = run.check_sequencing_status()
         if not sequencing_done:
             run.status = "sequencing"
-            if run.status_changed:
+            if run.status_changed():
```
Comment (kedhammar) [inline taca/analysis/analysis_element.py:42]: I think(?) this is a bugfix

**Comment by kedhammar** [comment]
WIP

**Comment by kedhammar** [comment]
Finally got the test mocks to work both on local and GHA 😤 I believe it's ready to merge.

The changes of the source code only consist of what I believe to be a bugfix and minor formatting, the rest is just test-stuff which appears to be working now.

File: taca/analysis/analysis_element.py (hunk 1 above)
Comment (alneberg) [reply taca/analysis/analysis_element.py:42]: Yes I think you're right. But my search found 5 other `run.status_changed` without the function call...

Comment (kedhammar) [reply taca/analysis/analysis_element.py:42]: [05979c7](https://github.com/ssjunnebo/TACA/pull/21/commits/05979c7208b377d60d569a3a79512639ea84e53f)

**Review by alneberg — APPROVED** [review]
I think it looks good and I agree with the bugfix! Thank you!

File: taca/analysis/analysis_element.py (hunk 1 above)
Comment (alneberg) [reply taca/analysis/analysis_element.py:42]: Thank you!

**Comment by kedhammar** [comment]
@alneberg OK to set required python version in setup.py? I found it a bit weird that it wasn't explicated anywhere. It's the same version that is used by TACA on preproc now.

[5599bed](https://github.com/ssjunnebo/TACA/pull/21/commits/5599bed3fb9b2ead09bcfd14dfd418762db258e2)
[488cb39](https://github.com/ssjunnebo/TACA/pull/21/commits/488cb39f01e2d2806ae1a5c351f0b16d6fb07cdb)

**Comment by alneberg** [comment]
If it works I guess that is fine!

**Review by alneberg — APPROVED** [review]
as before

**Review by ssjunnebo — APPROVED** [review]
Thanks!

**Pull request merged**